app/models/support/requests/brexit_feedback_request.rb

```diff
@@ -0,0 +1,78 @@
+require 'browser'
+require 'time'
+
+module Support
+  module Requests
+    class BrexitFeedbackRequest < Request
+      attr_accessor :formatted_results
+      attr_reader :from_date, :to_date
+      def initialize(from_date, to_date, slugs)
+        @from_date = from_date
+        @to_date = to_date
+        @brexit_slugs = slugs
+        @support_api = support_api
+        @first_page = get_first_page
+        @formatted_results = format_results(@first_page).reject do |res|
+          is_brexity?(res["path"]) == false
+        end
+        get_subsequent_pages
+      end
+
+      def support_api
+        GdsApi::SupportApi.new(
+          Plek.new.find("support-api"),
+          bearer_token: ENV['SUPPORT_API_BEARER_TOKEN'],
+          timeout: 30,
+        )
+      end
+
+      def get_first_page
+        @support_api.problem_reports(from_date: @from_date, to_date: @to_date).to_hash
+      end
+
+      def get_subsequent_pages
+        page = 2
+        until page >= @first_page["pages"] + 1 do
+          results = @support_api.problem_reports(from_date: @from_date, to_date: @to_date, page: page).to_hash
+          @formatted_results += format_results(results)
+          page += 1
+        end
+      end
+
+      def format_results(result_hash)
```

Comment:
I don't think this belongs here. It's a response formatter and would be better split out into a separate class.

Reply:
👍